awx/ui_next/src/index.jsx

```diff
@@ -56,31 +55,38 @@ export function main(render) {
       }) => <Redirect to={`${pathname.slice(0, -1)}${search}${hash}`} />}
     />
   );
-  const loginRoutes = (
-    <Switch>
-      {removeTrailingSlash}
-      <Route
-        path="/login"
-        render={() => <Login isAuthenticated={isAuthenticated} />}
-      />
-      <Redirect to="/login" />
-    </Switch>
-  );
+
+  const defaultRedirect = () => {
+    if (isAuthenticated(document.cookie)) {
+      return <Redirect to="/home" />;
+    }
+    return (
+      <Switch>
+        {removeTrailingSlash}
+        <Route
+          path="/login"
+          render={() => <Login isAuthenticated={isAuthenticated} />}
+        />
+        <Redirect to="/login" />
+      </Switch>
+    );
+  };
 
   return render(
     <RootProvider>
       <I18n>
         {({ i18n }) => (
           <Background>
-            {!isAuthenticated(document.cookie) ? (
-              loginRoutes
-            ) : (
-              <Switch>
-                {removeTrailingSlash}
-                <Route path="/login" render={defaultRedirect} />
-                <Route exact path="/" render={defaultRedirect} />
-                <Route
-                  render={() => (
+            <Switch>
+              {removeTrailingSlash}
+              <Route path="/login" render={defaultRedirect} />
+              <Route exact path="/" render={defaultRedirect} />
+              <Route
+                render={() => {
+                  if (!isAuthenticated(document.cookie)) {
```

Comment:
👍 Very cool. I like this method of conditionally routing/rendering within the render func.